### ISSUE TITLE
Some Elements Are Not Being Parsed Into Strings

### ISSUE THREAD

**Issue by drj826** [drj826/SML#43]
I recently changed the parsing logic of when blocks are parsed into strings.  Formerly blocks were parsed into strings at the end of the parse method.  I changed it so blocks are parsed into strings when the parser reaches the end of a document.

How do I know they're not being parsed into strings?  They appear in the published document as a raw element like, "title:: Semantic Manuscript Language (SML)".

After making this change I noticed certain elements were no longer parsed into strings.  For example:

- some source elements in a document list of references
- allocation title components
- entity descriptions on the entities page
- entity descriptions in the entities themselves
- there are probably others...

**Comment by drj826** [comment]
Here's an interesting clue.  In the library list of references two of the sources DO NOT exhibit this problem of unparsed blocks: 'Lamport96' and 'chicago'.

What do these two have in common?  They are in the sml.txt file directly and are not included via an 'include::' element.  I don't understand why this would make a difference.  I think the key to fixing this issue is understanding this.

**Comment by drj826** [comment]
I made a change that partially fixed the problem.  Source elements are now parsed and presented correctly.  What did I change?

It turns out I was trying to be too clever in the SML::Parser '_resolve_include_line' private method.  I was unnecessarily using recursion to retrieve the list of lines to be included.  I was invoking the parser to get the included division and then asking the division for its lines.  I simplified the implementation to use the parser's '_get_line_list_for_id' and it partially fixed this issue.

I suspect the rest of the issue can be fixed (for entity elements) by making a similar change somewhere else in the code.
